backend/src/layer/data/file_handler/mod.rs

```diff
@@ -0,0 +1,92 @@
+//! See [`FileHandler`].
+
+use std::path::PathBuf;
+
+use crate::util::IMAGE_EXTENSION;
+
+use async_trait::async_trait;
+use tracing::trace;
+
+use crate::{
+    interface::image_storage::{ImageStorage, Result},
+    util::{ImageResource, Uuid},
+};
+
+/// Struct containing all information necessary to construct a [`FileHandler`].
+pub struct FileHandlerInfo {
+    /// Path where images should be stored
+    pub image_path: PathBuf,
+}
+
+/// Class for saving images to the file system.
+pub struct FileHandler {
+    image_path: PathBuf,
+}
+
+impl FileHandler {
+    #[must_use]
+    pub fn new(info: FileHandlerInfo) -> Self {
+        Self {
+            image_path: info.image_path,
+        }
+    }
+}
+
+#[async_trait]
+impl ImageStorage for FileHandler {
+    async fn save_image(&self, id: Uuid, image: ImageResource) -> Result<()> {
+        let mut file_path = self.image_path.clone();
+        file_path.push(id.to_string());
+        file_path.set_extension(IMAGE_EXTENSION);
+
+        tokio::task::spawn_blocking(move || image.save(file_path))
+            .await
+            .expect("image saving should not panic nor get aborted")?;
+
```

Comment:
If an image could not be saved (with `image.save(file_path)`), does the server crash? And if not, all operations that follow will be canceled? Is the image_id then referring to nothing? Does the user/client get an error message?

Reply:
Which reverse seems to be easier? Or should this case be ignored?

Reply:
Probably easier to remove database entry, but in any case, this has to be taken care of in the command layer